cpp/include/raft/matrix/detail/gather.cuh

```diff
@@ -350,20 +351,23 @@ void gather_if(const InputIteratorT in,
  * Helper function to gather a set of vectors from a (host) dataset.
  */
 template <typename T, typename IdxT, typename MatIdxT = int64_t>
-void gather_buff(host_matrix_view<const T, MatIdxT> dataset,
+void gather_buff(raft::resources const& res,
+                 host_matrix_view<const T, MatIdxT> dataset,
                  host_vector_view<const IdxT, MatIdxT> indices,
                  MatIdxT offset,
                  pinned_matrix_view<T, MatIdxT> buff)
 {
-  raft::common::nvtx::range<common::nvtx::domain::raft> fun_scope("gather_host_buff");
+  auto dim        = buff.extent(1);
   IdxT batch_size = std::min<IdxT>(buff.extent(0), indices.extent(0) - offset);
+  raft::common::nvtx::range<common::nvtx::domain::raft> fun_scope(
+    "gather_host_buff(%zu, %zu)", uint64_t(batch_size), uint64_t(dim));
 
 #pragma omp for
   for (IdxT i = 0; i < batch_size; i++) {
     IdxT in_idx = indices(offset + i);
-    for (IdxT k = 0; k < buff.extent(1); k++) {
-      buff(i, k) = dataset(in_idx, k);
-    }
+    raft::copy(res,
```

Comment:
So the resource is actually unused since the src and dest are both in host mem? I wonder if we can create a resource on the fly and supply it to `raft::copy()` rather than adding the resource to `gather_buff`'s signature.